R/R6.R

```diff
@@ -93,45 +99,57 @@ neo4j_api <- R6::R6Class(
         user = self$user
       )
     },
+    # In 4.0 the endpoints have changed.  Instead we will call
+    # the appropriate cypher or function
+    #
     # Test if the endpoint is accessible
     # This only return the status code for now so I wonder
     # if we should make if verbose instead of just returning SC
     ping = function() {
       # browser()
-      attempt(status_code(get_wrapper(self, "db/data/relationship/types")))
+      res<-'call db.ping()' %>% call_neo4j(self, type='row')
+      #attempt(status_code(get_wrapper(self, "db/data/relationship/types")))
     },
     # Get Neo4J version
     get_version = function() {
-      res <- get_wrapper(self, "db/data")
+      #hitting the base url should reply back with some version info
+      if (self$isV4 == TRUE ) {
+        turl <- ""
+      } else {
+        turl <- "db/data"
+      }
+      res <- get_wrapper(self, turl)
       content(res)$neo4j_version
     },
+    # Get Neo4J cluster status
+    #cluster_status = function() {
+      #cluster doesn't work hmmm
+     # res <- get_wrapper(self, glue('db/{self$db}/cluster/status'))
+    #},
     # Get a list of relationship registered in the db
     # return it as a data.frame because data.frame are cool
     get_relationships = function() {
-      res <- get_wrapper(self, "db/data/relationship/types")
-      tibble(labels = as.character(content(res)))
+      res<-'call db.relationshipTypes()' %>% call_neo4j(self, type='row')
     },
     # Get a list of labels registered in the db
     # Tibbles are awesome
     get_labels = function() {
-      res <- get_wrapper(self, "db/data/labels")
-      tibble(labels = as.character(content(res)))
+      # call db.labels()
+      res<-'call db.labels()' %>% call_neo4j(self, type='row')
     },
     get_property_keys = function() {
-      res <- get_wrapper(self, "db/data/propertykeys")
-      tibble(labels = as.character(content(res)))
+      #Make CYPHER CALL
+      res<-'call db.propertyKeys()' %>% call_neo4j(self, type='row')
```

Comment:
Will these still work with Neo4J < 4.0.0 (i.e older versions of Neo4J)? 

We need to stay conservative in the way this package behaves, so that we are sure we don't break old code with these changes.